### PR TITLE
test(x/twap): 3-asset geometric logic tests

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #3443

## What is the purpose of the change

Adding 3-asset geometric twap logic tests as per issues

### CODE REVIEW

File: x/twap/logic_test.go [hunk 1]
```diff
@@ -1558,6 +1592,33 @@ func (s *TestSuite) TestAfterCreatePool() {
 	}
 }
 
+func (s *TestSuite) TestTwapLog() {
```
Comment (p0mvn) [inline x/twap/logic_test.go:1595]: Note: these are moved to group tests at the top of the file and helpers at the bottom. These are unchanged

File: x/twap/keeper_test.go [hunk 2]
```diff
@@ -583,6 +583,7 @@ func newThreeAssetOneSidedRecord(time time.Time, accum sdk.Dec, useP0 bool) []ty
 	} else {
 		record.P1ArithmeticTwapAccumulator = accum
 	}
+	record.GeometricTwapAccumulator = accum
```
Comment (ValarDragon) [inline x/twap/keeper_test.go:586]: odd that its the same accumulator set for arithmetic and geometric ones, but w/e

File: x/twap/logic_test.go [hunk 3]
```diff
@@ -791,6 +780,51 @@ func TestComputeArithmeticTwap_ThreeAsset(t *testing.T) {
...
 
+func TestComputeArithmeticTwap_ThreeAsset_Geometric(t *testing.T) {
+	var (
+		five        = sdk.NewDec(5)
+		fiveFor3Sec = OneSec.MulInt64(3).Mul(twap.TwapLog(five))
+
+		ten          = five.MulInt64(2)
+		tenFor100Sec = OneSec.MulInt64(100).Mul(twap.TwapLog(ten))
+
+		errTolerance = sdk.MustNewDecFromStr("0.00000001")
```
Comment (ValarDragon) [inline x/twap/logic_test.go:791]: How is this chosen?

File: x/twap/logic_test.go (hunk 1 above)
Comment (ValarDragon) [reply x/twap/logic_test.go:1595]: Ok... , don't think this is an important code style thing though

File: x/twap/logic_test.go (hunk 3 above)
Comment (ValarDragon) [reply x/twap/logic_test.go:791]: Can we make an issue to revisit this error tolerance in the lens of stableswap with high scaling factors?

**Review by ValarDragon — APPROVED** [review]
LGTM, but need to revisit the precision pre-release

File: x/twap/logic_test.go (hunk 3 above)
Comment (p0mvn) [reply x/twap/logic_test.go:791]: This was chosen as the tightest allowed precision with the current `Pow` function. Should be possible to make tighter once the new `Power` function is merged.

Will make an issue

Comment (p0mvn) [reply x/twap/logic_test.go:791]: https://github.com/osmosis-labs/osmosis/issues/3770

**Comment by p0mvn** [comment]
Issue created: https://github.com/osmosis-labs/osmosis/issues/3770

Going to merge with one approval since this is a test and < 100 LOC

**Pull request merged**